src/type_translator.ts

```diff
@@ -530,10 +522,9 @@ export class TypeTranslator {
         return '?';
       }
       const name = this.symbolToString(type.symbol, /* useFqn */ true);
-      if (name === '(Anonymous class)') {
-        // Values that have anonymous class types produce this name, but the type
-        // appears otherwise identical to a named class.  Given that the type is
-        // anonymous here, there's not really a useful name we can emit.
+      if (name === '?') {
+        // Values that have anonymous class types produce '?'. Make sure not to emit '!?', which is
```

Comment:
I'm a little surprised by this change.  I can see that by not using some user-facing 'display builder' we'd no longer get the '(Anonymous class)' string, but I can't see why the above code ever produces the '?' string

Reply:
see the `if (!name) return '?';` line just after calling `symbolToEntityName`. That API returns `undefined` for things that cannot be properly named, which we turn into `?`.